jsapp/js/components/modalForms/formMedia.es6

```diff
@@ -37,20 +31,46 @@ class FormMedia extends React.Component {
    */
 
   componentDidMount() {
-    this.refreshFormMedia();
+    const callbacks = {
+      onComplete: this.onGetMediaCompleted.bind(this),
+      onFail: this.onGetMediaFailed.bind(this)
+    };
```

Comment:
It would be simpler and genuinely better if instead of passing `callbacks` object (and handling them in `actions.es6`), you would simply listen to the `completed` and `failed` built-in callbacks. Generally `onComplete` and `onFail` custom callbacks are a bad code and we should use them only when regular actions flow is for some reason too hard to do.

So you would just need to change these callback lines to something like:

```
actions.media.loadMedia.completed.listen(this.onGetMediaCompleted);
actions.media.loadMedia.failed.listen(this.onGetMediaFailed);
```

Reply:
First attempt had a lot of :spaghetti:! I think I learned from this for `#2928` :crossed_fingers: